### ISSUE TITLE
Can't checkout saleor repo

### ISSUE THREAD

**Issue by yoxud** [saleor/saleor#7135]
### What I'm trying to achieve
Locally clone a fork of Saleor

### Steps to reproduce the problem
1.Fork mirumee/saleor
2. Locallly execute git clone myfork

### What I expected to happen
Expected the checkout process to succeed but instead got the error
`fatal: cannot create directory at 'saleor/graphql/core/tests/cassettes/test_get_oembed_data[http:': Invalid argument
warning: Clone succeeded, but checkout failed.`
Seems to be due to the colon in the path name as colons are not acceptable characters for Windows paths...
The error is also present with another neigbourhing path using https instead of http.

### Screenshot
Here is a screenshot of one of the malformed path names
![mirumee](https://user-images.githubusercontent.com/41711846/114230699-76cff500-9947-11eb-8765-1994b9312765.jpg)

**Comment by krzysztofwolski** [comment]
Thank you for reporting!

**Comment by auvipy** [comment]
can we checkout now?

**Comment by yoxud** [comment]
nope. same error.

**Comment by Megidd** [comment]
I'm experiencing this issue in #7279 with version `2.11.0`.

**Comment by p-zauner** [comment]
still same error

**Comment by Megidd** [comment]
The error is still thrown. Is there anyway to tweek Windows 10 to allow such file path strings?

**Comment by Megidd** [comment]
This might help: https://stackoverflow.com/q/63727594/3405291

**Comment by yoxud** [comment]
@Megidd  : Not to my knowledge. I ended up installing an ubuntu vm and building it from the vm.

**Comment by Megidd** [comment]
Thanks @yoxud  

The option I'm going to try is:

1. Fork the repository 
1. Clone the fork on Linux
1. Change the path names on Linux
1. Commit and push
1. Then check out the fixed forked repository on Windows

**Comment by SurendraTamang** [comment]
Any progress on this issue can we pull on windows 10?

**Comment by NimishVerma** [comment]
Can not clone on Windows 10. Any fixes yet?  Deltas get resolved and then bad files throw error. 
```
Resolving deltas: 100% (139156/139156), done.
error: invalid path 'saleor/graphql/core/tests/cassettes/test_get_oembed_data[http:/www.youtube.com/watch?v=dQw4w9WgXcQ-VIDEO].yaml'
fatal: unable to checkout working tree
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status
and retry with 'git restore --source=HEAD :/'
```

**Comment by D1ll1ng** [comment]
Still no news? Windows users cant use Saleor right now, and haven't have not been able to in the last 2 months. Is possible for someone to make a new repository of Saleor perhaps the latest version 3.0 and post it for us to clone :) without the rick roll please...

**Comment by lioden** [comment]
@D1ll1ng haven't tried to run tests on it but might work
[8c93013](https://github.com/lioden/saleor/commit/8c930137375f3ca16fe49dcbe007851e8e72c3be)

**Comment by lioden** [comment]
@D1ll1ng I created a fork without the cassettes folder, using it as a temporary solution ☝️

**Comment by maihungdev** [comment]
For those who want just the commands to workaround the issue:

git clone --sparse -c core.protectNTFS=false -n <repo-URL>
git sparse-checkout add "\!<pattern1>" "\!<pattern2>"
git checkout <branch>
Patterns are relative to repo root and can use *

**Comment by maihungdev** [comment]
Finaly, i fixed it! 

1. Delete saleor folder
2. Clone sub repository in saleor-platform folder, cmd: git clone --sparse -c core.protectNTFS=false -n https://github.com/mirumee/saleor
3. cd to saleor folder: cmd: git sparse-checkout add "\!<pattern1>" "\!<pattern2>"
4. git checkout master

**Comment by ohuu** [comment]
I've just hit this problem. It looks like the `[` and `]` characters in the filename that are causing Windows problems. Is it not possible to just rename these files? This is a pretty serios problem for devs using Windows and it seems like a pretty straight-forward fix, just wondering if there's something stopping the Saleor devs from doing it?

Thanks @lioden for forking the project 👍

**Comment by maarcingebala** [comment]
Fixed in #9236
